text/0000-optional-jquery.md

```diff
@@ -0,0 +1,248 @@
+events or assert the state of the DOM is still the standard, based on `this.$()` returning a jQuery object representing
+the rendered result of the tests `render` call.
+
+To be able to reliably run tests in a jQuery-less world, we need to run our tests without jQuery being included,
+so our test harness has to work without jQuery as well.
+
+Fortunately this is well underway already. [ember-native-dom-helpers](https://github.com/cibernox/ember-native-dom-helpers)
+introduced native DOM test helpers for integration and acceptance tests as an user space addon. The recent acceptance 
+testing [RFC 268](https://github.com/emberjs/rfcs/blob/master/text/0268-acceptance-testing-refactor.md) provides 
+similar test helpers, implemented in the `@ember/test-helpers` package, and envisages deprecating the global test 
+helpers.
+
+However while the existing jQuery based APIs are still available, when these are used without jQuery they have to throw
+an assertion with some meaningful error message:
+
+* global acceptance test helpers that expect jQuery selectors (which are a potentially incompatible superset of standard
+CSS selectors)
+
+* `this.$()` in component tests, provided currently by `@ember/test-helpers` in `moduleForComponent` and 
+`setupRenderingTest`
+
+In both cases the error message should state that jQuery is not available and that the native DOM based test helpers
+of the `@ember/test-helpers` package should be used instead.
+
+The transitioning to these new test helpers can be eased through a codemod. For `ember-native-dom-helpers` there already
+exists  [ember-native-dom-helpers-codemod](https://github.com/simonihmig/ember-native-dom-helpers-codemod), which 
+could be adapted to the very similar RFC 268 based interaction helpers in `@ember/test-helpers`.
+
+### Implementation outline
+
+The following outlines how a possible implementation of the jQuery integration flag *could* look like. This
+is just to provide some additional context, but is *intentionally not* meant to be normative, to allow some flexibility 
+for the actual implementation.
+
+The addon that will handle the flag is expected to be [ember-optional-features](https://github.com/emberjs/ember-optional-features),
+which will read from and write to a `config/optional-features.{js,json}` file. This will hold the `jquery-integration` 
+flag (amongst others). This flag in turn will be added to the `EmberENV` hash, which will make Ember go into its 
+"no jQuery" mode when set to `false`. 
+
+Ember CLI and the `@ember/jquery` addon will also look for `jquery-integration` in this configuration file, and will 
+opt-out of importing jQuery when this file is present and the flag is set to `false`.
+
+## How we teach this
+
+### Guides
+
+The existing "Managing Dependencies" chapters in the Ember Guides as well as on ember-cli.com provide a good place to 
+explain users how to set the jQuery integration flag by means of the mentioned addon that handles this flag. 
```

Comment:
Clarify this is not the `@ember/jquery` addon

Reply:
Yes, great point. Currently we are thinking that it will be `@ember/optional-features` ([WIP repo here](https://github.com/emberjs/ember-optional-features))...